internal/server/coap/coap.go

```diff
@@ -51,13 +56,31 @@ func (s *Server) Start() error {
 		if err != nil {
 			return fmt.Errorf("failed to load auth certificates: %w", err)
 		}
-		tlsConfig := &tls.Config{
-			Certificates: []tls.Certificate{certificate},
+		dtlsConfig := &piondtls.Config{
+			Certificates:         []tls.Certificate{certificate},
+			ExtendedMasterSecret: piondtls.RequireExtendedMasterSecret,
+			ClientAuth:           piondtls.RequireAndVerifyClientCert,
+			ConnectContextMaker: func() (context.Context, func()) {
+				return context.WithTimeout(s.Ctx, 30*time.Second)
+			},
+		}
+		clientCA, err := loadCertFile(s.Config.ClientCAFile)
+		if err != nil {
+			return fmt.Errorf("failed to load client ca file: %w", err)
+		}
+		if len(clientCA) > 0 {
+			if dtlsConfig.ClientCAs == nil {
+				dtlsConfig.ClientCAs = x509.NewCertPool()
+			}
+			if !dtlsConfig.ClientCAs.AppendCertsFromPEM(clientCA) {
+				return fmt.Errorf("failed to append client ca to tls.Config")
+			}
 		}
 
 		go func() {
-			errCh <- gocoap.ListenAndServeTCPTLS("udp", s.Address, tlsConfig, s.handler)
+			errCh <- gocoap.ListenAndServeDTLS("udp", s.Address, dtlsConfig, s.handler)
```

Comment:
why is ListenTCP removed, we shoulld have all 3 options, no TLS, TLS and DTLS. Modify the config file and this